clients/include/testing_csrmm.hpp

```diff
@@ -335,65 +393,6 @@ hipsparseStatus_t testing_csrmm(Arguments argus)
     }
     int nnz = M * scale * K;
 
-    // Argument sanity check before allocating invalid memory
-    if(M <= 0 || N <= 0 || K <= 0)
```

Comment:
When k == 0 and M and N > 0 we enter here (even this combination of arguments is in fact valid). We then call the routine using arrays that are not properly allocated (we use safe_size instead of the size they should actually be). Inside rocsparse, when beta is non-zero we still need to scale C = beta * C and this is where we write out of bounds. Solution like in rocsparse is to not do this kind of invalid arguments checks in the valid args section of the testing. This simplifies the code and fixes the issue with no loss in test coverage.